### PR TITLE
Improve threading when calling the accessy api.

### PR DESCRIPTION
**Pull request by HalfVoxel**
The previous code started one thread per member, which is way too much, and just resulted
in a ton of rate limit errors.

## Summary by CodeRabbit

- **New Features**
	- Enhanced request handling logic for improved performance during high traffic.
	- Optimized threading model for user data retrieval, allowing for more efficient processing of user IDs.

- **Bug Fixes**
	- Adjusted backoff strategy for handling HTTP 429 responses to improve retry behavior.

### CODE REVIEW

**Review by emanuelen5 — APPROVED** [review]
LGTM. Not sure how the backoff affects it, but if it works, it works.

**Comment by emanuelen5** [comment]
I fixed the build so you can now rebase on main to get the tests running.

File: api/src/multiaccessy/accessy.py [hunk 1]
```diff
@@ -541,14 +542,23 @@ def fill_membership_id(user: AccessyMember) -> None:
             user.membership_id = data["id"]
 
         threads = []
+        user_ids = list(user_ids)
+        thread_count = min(4, len(user_ids))
```
Comment (BerglundDaniel) [inline api/src/multiaccessy/accessy.py:546]: I don't follow this completely. Doesn't this still create one thread per user_id as before?

Comment (HalfVoxel) [reply api/src/multiaccessy/accessy.py:546]: No, it creates at most 4 threads.

**Pull request merged**